Unable to install -- cp: cmark-gfm/build/src/libcmark-gfm.so*: No such file or directory

I think an issue might have cropped up which has caused this shard to fail postinstall. I believe it to be nonfunctional on a mac. Here's why:

I'm attempting to run @mamantoha's https://github.com/mamantoha/shards-info locally, and I have the latest master branch. It points to the cr-cmark-gfm master branch.

When I run `shards update`, it fails the postinstall step during the final `cp` steps which move the .so files into place. 


```
<snip>
[ 98%] Building C object api_test/CMakeFiles/api_test.dir/main.c.o
[ 98%] Built target cmark-gfm
[100%] Linking CXX executable api_test
[100%] Built target api_test
Binaries can be found in build/src
cp cmark-gfm/build/src/libcmark-gfm.a .
cp cmark-gfm/build/src/libcmark-gfm.so* .
cp: cmark-gfm/build/src/libcmark-gfm.so*: No such file or directory
make: *** [build] Error 1
```

I looked for some missing library, but the compilation appears to finish successfully, and I observe no error messages from that part of the process. 

Moreover, I cloned down https://github.com/github/cmark-gfm directly (at tag: 0.29.0.gfm.8) and I'm able to compile it with just `make cmake_build`. The output from that compilation is:

```
<snip>
[ 97%] Linking C executable cmark-gfm
[ 98%] Building C object api_test/CMakeFiles/api_test.dir/main.c.o
[ 98%] Built target cmark-gfm
[100%] Linking CXX executable api_test
[100%] Built target api_test
Binaries can be found in build/src

cmark-gfm> ls build/src
total 1488
drwxr-xr-x  13 robert  staff     416 Sep 25 22:50 .
drwxr-xr-x  13 robert  staff     416 Sep 25 22:50 ..
drwxr-xr-x   9 robert  staff     288 Sep 25 22:50 CMakeFiles
-rw-r--r--   1 robert  staff   50067 Sep 25 22:50 Makefile
-rw-r--r--   1 robert  staff    5952 Sep 25 22:50 cmake_install.cmake
-rwxr-xr-x   1 robert  staff   35800 Sep 25 22:50 cmark-gfm
-rw-r--r--   1 robert  staff    1082 Sep 25 22:50 cmark-gfm_export.h
-rw-r--r--   1 robert  staff     179 Sep 25 22:50 cmark-gfm_version.h
-rw-r--r--   1 robert  staff    1393 Sep 25 22:50 config.h
-rwxr-xr-x   1 robert  staff  267184 Sep 25 22:50 libcmark-gfm.0.29.0.gfm.6.dylib
-rw-r--r--   1 robert  staff  376344 Sep 25 22:50 libcmark-gfm.a
lrwxr-xr-x   1 robert  staff      31 Sep 25 22:50 libcmark-gfm.dylib -> libcmark-gfm.0.29.0.gfm.6.dylib
-rw-r--r--   1 robert  staff     314 Sep 25 22:50 libcmark-gfm.pc
```

I suspect this is because the output from the compiler isn't a `.so` file, but instead a `.dylib` -- however, it's been far too many years since I wrote any C to have any confidence in that guess. And I've never actually worked with crystal bindings to shared libraries, so I have no idea what is actually needed here.

--- a/ext/Makefile
+++ b/ext/Makefile
@@ -1,14 +1,20 @@
 CMARK = cmark-gfm
 SOURCES = $(wildcard $(CMARK)/src/*.c) $(wildcard $(CMARK)/src/*.h)
+COMPILED_EXT = so
+
+OS := $(shell uname)
+ifeq ($(OS),Darwin)
+	COMPILED_EXT = dylib
+endif
 
 all: build
 
 build: $(CMARK) $(SOURCES)
 	cd $(CMARK) && INSTALL_PREFIX=.. make cmake_build
 	cp $(CMARK)/build/src/libcmark-gfm.a .
-	cp $(CMARK)/build/src/libcmark-gfm.so* .
+	cp $(CMARK)/build/src/libcmark-gfm.$(COMPILED_EXT)* .
 	cp $(CMARK)/build/extensions/libcmark-gfm-extensions.a .
-	cp $(CMARK)/build/extensions/libcmark-gfm-extensions.so* .
+	cp $(CMARK)/build/extensions/libcmark-gfm-extensions.$(COMPILED_EXT)* .
 
 $(CMARK):
 	git clone --depth 1 --branch 0.29.0.gfm.8 https://github.com/github/cmark-gfm.git
@@ -18,4 +24,4 @@ clean:
 
 distclean: clean
 	rm -rf *.a
-	rm -rf *.so*
+	rm -rf *.$(COMPILED_EXT)*

